servers/rendering/rendering_device.cpp

```diff
@@ -181,13 +181,34 @@ Ref<RDShaderSPIRV> RenderingDevice::_shader_compile_spirv_from_source(const Ref<
 
 	Ref<RDShaderSPIRV> bytecode;
 	bytecode.instantiate();
-	for (int i = 0; i < RD::SHADER_STAGE_MAX; i++) {
-		String error;
+	String error;
```

Comment:
It is mandated in the newer GLSL that you specify #version so shaders will probably never be empty if they exist. Seems like its easier to just check for empty shader and continue if nothing found.